modules/geolocation/geolocation.py

```diff
@@ -5,6 +5,8 @@
 import numpy as np
 import math
 import logging
+import os.path
```

Comment:
How necessary is this?

Additionally, don't forget to merge/rebase main into this branch.